pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -22,6 +22,13 @@ package pulsar.proto;
 option java_package = "org.apache.pulsar.common.api.proto";
 option optimize_for = LITE_RUNTIME;
 
+message Schema {
+    required string name = 1;
+    required bytes version = 2;
+    required bytes schema_data = 3;
+    repeated KeyValue properties = 4;
```

Comment:
I think it was already pointed out, though it's missing now. 

 1. Shouldn't `version` be a `long` type?
 2. Since we are not supporting arbitrary schemas, I think we should identify the type of the schema that the client is sending. We should have an enum of choices, but use `int` type on the wire to prepare for when adding new enum values.

Reply:
Re: version, I have no way of telling what the backend will use to represent a version. If we pigeon hole implementations into using a long then we could force pulsar to store extra state to convert between a "pulsar version" and a "storage version". Additionally, I reasoned that you either receive a version string from an API, for example the REST endpoint, or through a consumed event, like through the consumer. This data does not need to be human readable but we can think of schemes to make it so (hash, encoded, etc).

Reply:
re: type. I will acquiesce. I prefer the "open" implementation but recognize that the current set of types is limited and growing slowly or not at all.

Reply:
Though I think that, when we attach to a message, the `schema_version` field is anyway defined as `int64`